### ISSUE TITLE
Limit cannon angle based on height

### ISSUE THREAD

**Issue by arouinfar** [phetsims/projectile-motion#53]
In a recent design meeting, we brought up the issue of handling projectiles shot from low height/shallow angle. One way we could address this issue is by limiting the cannon angle based on the pedestal height. 

For a height of 3m or less, the cannon's angle could be limited so that it doesn't point into the ground (as in the left column). Screenshots in the middle column illustrate the proposed limit. The cannon runs into a barrier at each limit, so it should feel natural to the user. Shooting a large projectile at these limits also has a reasonable trajectory (right column). 
<img src="https://user-images.githubusercontent.com/8419308/27611349-1651c9ea-5b4f-11e7-9780-ad0e246af246.png" width=200> <img src="https://user-images.githubusercontent.com/8419308/27611365-24d21704-5b4f-11e7-94c0-1802713d7762.png" width=200>  <img src="https://user-images.githubusercontent.com/8419308/27611895-5123b950-5b51-11e7-810e-22c8a3e42ba4.png" width=200>

<img src="https://user-images.githubusercontent.com/8419308/27611419-53044ef8-5b4f-11e7-87a5-751ac523e48f.png" width=200>  <img src="https://user-images.githubusercontent.com/8419308/27611439-6b980996-5b4f-11e7-871e-738a2cf74b9a.png" width=200>  <img src="https://user-images.githubusercontent.com/8419308/27611857-209e7572-5b51-11e7-9e0a-964b8db4350f.png" width=200>

<img src="https://user-images.githubusercontent.com/8419308/27611484-8eb1fcc0-5b4f-11e7-9108-bce57545b0ae.png" width=200>  <img src="https://user-images.githubusercontent.com/8419308/27611515-aa78e6f8-5b4f-11e7-8619-6ed40a882d7b.png" width=200> <img src="https://user-images.githubusercontent.com/8419308/27611829-00f39d74-5b51-11e7-8eaf-bdc050480ffe.png" width=200>

<img src="https://user-images.githubusercontent.com/8419308/27611570-f05f7b28-5b4f-11e7-80f0-22de14f29978.png" width=200> <img src="https://user-images.githubusercontent.com/8419308/27611550-d8ba4c14-5b4f-11e7-826d-7f601b347ce9.png" width=200> <img src="https://user-images.githubusercontent.com/8419308/27611795-d83dc3b4-5b50-11e7-8efd-00c926519f13.png" width=200>

Heights of 4m and above accommodate an angle of -90* without the cannon running into a natural barrier. However, once #51 is addressed, the projectile has the potential to obscure the cannon body. At -90*, the Buick would completely obscure the cannon. If the angle is limited to -60*, the cannon would still be somewhat obscured, but the barrel could still be grabbed.
<img src="https://user-images.githubusercontent.com/8419308/27613363-63940a5c-5b58-11e7-9126-9705ca7448d4.png" width=300> <img src="https://user-images.githubusercontent.com/8419308/27613628-cf63add6-5b59-11e7-8c95-118e6c3fa79a.png" width=300>

Limiting the angle to -60* would be consistent with Flash. The cannon can reach -90*, but only by typing it into the field. The cannon itself cannot be dragged beyond -60*.

In summary, the minimum angle would be:
* 0 m: 25*
* 1 m: -5*
* 2 m: -20*
* 3 m: -40*
* 4+ m: -60*

Tagging for further discussion in design meeting.

**Comment by arouinfar** [comment]
While the cannon can be obscured by the projectile, the cannon barrel can still be easily grabbed and adjusted. So perhaps we can allow the full range of angle for all heights > 3 m.

![image](https://user-images.githubusercontent.com/8419308/27656756-6d12c0be-5c07-11e7-8a49-4bcee6fb7e5e.png)  ![image](https://user-images.githubusercontent.com/8419308/27656875-ebde6042-5c07-11e7-8c53-b3f388b12d95.png)

**Comment by andrea-phet** [comment]
The above commit limits the cannon based on specs so far.

**Comment by arouinfar** [comment]
Thanks @andrea-phet! It's really helpful seeing the limit in action. If we decide to go this route, we'll need to come up with a solution for a situation like this:
![image](https://user-images.githubusercontent.com/8419308/27661216-b62316aa-5c16-11e7-9b31-9b4a80150df8.png)

Suppose the cannon was adjusted to -90* before being lowered into the ground. How would we handle that? The cannon could automatically rotate to the maximum allowed angle as the cannon is lowered, like so:

![image](https://user-images.githubusercontent.com/8419308/27661274-e3fa9f58-5c16-11e7-8528-b33bfa23567a.png) 
![image](https://user-images.githubusercontent.com/8419308/27661285-f1b015b0-5c16-11e7-85b7-265ed33229eb.png)
![image](https://user-images.githubusercontent.com/8419308/27661302-ff63e6a0-5c16-11e7-9b98-0c2d34d72a14.png)
![image](https://user-images.githubusercontent.com/8419308/27661309-08b83422-5c17-11e7-8bcf-d6eae533c346.png)
![image](https://user-images.githubusercontent.com/8419308/27661319-135496f0-5c17-11e7-9966-65e319bb2064.png)

**Comment by andrea-phet** [comment]
I think it looks really good with that update!

**Comment by arouinfar** [comment]
Love it!!! Let's review in design meeting, but I think this will take care of the issue.

**Comment by arouinfar** [comment]
Reviewed in design meeting. 👍
